server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -43,6 +44,7 @@ public class CombinedDeletionPolicy extends IndexDeletionPolicy {
     private final SoftDeletesPolicy softDeletesPolicy;
     private final LongSupplier globalCheckpointSupplier;
     private final Map<IndexCommit, Integer> acquiredIndexCommits; // Number of references held against each commit point.
+    private final Set<IndexCommit> indexCommitsAcquiredByCommitsListener;
```

Comment:
A comment on this explaining it's purpose would help the next reader.

Reply:
I've added a commit.